lib/ansible/modules/network/cloudengine/ce_facts.py

```diff
@@ -0,0 +1,440 @@
+#!/usr/bin/python
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = """
+---
+module: ce_facts
+version_added: "2.3"
+author: "JackyGao2016 (@CloudEngine-Ansible)"
+short_description: Gets facts about HUAWEI CloudEngine switches
+description:
+  - Collects facts from CloudEngine devices running the CloudEngine
+    operating system.  Fact collection is supported over Cli
+    transport.  This module prepends all of the base network fact keys
+    with C(ansible_net_<fact>).  The facts module will always collect a
+    base set of facts from the device and can enable or disable
+    collection of additional facts.
+extends_documentation_fragment: cloudengine
+
+options:
+  gather_subset:
+    description:
+      - When supplied, this argument will restrict the facts collected
+        to a given subset.  Possible values for this argument include
+        all, hardware, config, and interfaces.  Can specify a
+        list of values to include a larger subset.  Values can also be used
+        with an initial C(M(!)) to specify that a specific subset should
+        not be collected.
+    required: false
+    default: '!config'
+"""
+
+EXAMPLES = """
+# Note: examples below use the following provider dict to handle
+#       transport and authentication to the node.
+vars:
+  cli:
+    host: "{{ inventory_hostname }}"
+    username: admin
+    password: admin
+    transport: cli
+
+- ce_facts:
+    gather_subset: all
+
+# Collect only the config facts
+- ce_facts:
+    gather_subset:
+      - config
+
+# Do not collect hardware facts
+- ce_facts:
+    gather_subset:
+      - "!hardware"
+"""
+
+RETURN = """
+gather_subset:
+  description: The list of fact subsets collected from the device
+  returned: always
+  type: list
+
+# default
```

Comment:
FYI, all these RETURN items will end up as rows in a table and will not be grouped by default, hardware, etc.

I can't think of a better way of doing this, no action, just so you are aware.